src/datapilot/core/platforms/dbt/executor.py

```diff
@@ -158,3 +158,16 @@ def run(self):
                 self.logger.info(color_text(f"Skipping insight {insight_class.NAME} as {message}", YELLOW))
 
         return reports
+
+
+def main():
+    # Need to get the partial manifest or catalog here
+    manifest_path = "path/to/manifest.json"
```

Comment:
You might need to add logic of generating parital manifest +catalog

Reply:
Steps:
From pre-commit
- Get list of changed files. 
- Extract

Reply:
https://github.com/dbt-checkpoint/dbt-checkpoint/blob/main/dbt_checkpoint/utils.py#L665

Reply:
```
pre-commit hooks, as used in the context of software development, allow you to automate checks and tasks (like formatting and linting) before a commit is finalized in a version control system like Git. To get a list of changed files that a pre-commit hook can act upon, you would typically use Git commands within your hook script.

Here's a basic example of how you can get a list of changed files in a pre-commit hook script:

Create or edit your pre-commit hook: This is usually located in .git/hooks/pre-commit within your Git repository. If the file doesn't exist, you can create it. Make sure the script is executable.

Script to get changed files:
You can use Git commands within the script to get a list of staged files that are about to be committed. Here's a simple script that does this:

#!/bin/sh

# Get list of staged files
files=$(git diff --cached --name-only --diff-filter=ACM)

# Now you can loop over the files and perform checks
for file in $files; do
  echo "File changed: $file"
  # Add your logic here, e.g., run linters/formatters or custom checks on these files
done
In this script:

git diff --cached --name-only --diff-filter=ACM command is used to get a list of staged files (those that are Added (A), Copied (C), or Modified (M)).
The output is a list of file paths that have been staged for commit.
The script then loops through each file and prints its name. You can replace the echo command with any command you want to run for each file, such as linting or formatting.
Make the script executable: Before the script can run, you need to make sure it's executable. You can do this with the following command:
bash

chmod +x .git/hooks/pre-commit
Now, every time you attempt to make a commit, this pre-commit hook will run and print the list of files that have been changed and staged for commit. You can modify the script to include more sophisticated logic based on your project's needs.
```

Reply:
https://github.com/AltimateAI/vscode-dbt-power-user/pull/926/files#diff-ce5c8fffae5b21bfe8138ce41b467267b6d898014e6df125f67566c23f01e946R496